pkg/controllers/management/feature/feature_handler.go

```diff
@@ -46,9 +48,51 @@ func (h *handler) sync(_ string, obj *v3.Feature) (*v3.Feature, error) {
 	if obj.Name == features.Harvester.Name() {
 		return obj, h.toggleHarvesterNodeDriver(obj.Name)
 	}
+
+	if obj.Name == features.HarvesterBaremetalContainerWorkload.Name() {
+		return obj, h.syncHarvesterFeature(obj)
+	}
 	return obj, nil
 }
 
+// syncHarvesterFeature ensures that Harvester feature is enabled
+// if baremetal management feature is enabled and annotates feature with experimental annotation
+func (h *handler) syncHarvesterFeature(obj *v3.Feature) error {
+
+	objCopy := obj.DeepCopy()
+
+	if objCopy.Annotations == nil {
+		objCopy.Annotations = make(map[string]string)
+	}
+
+	if val, ok := objCopy.Annotations[v3.ExperimentalFeatureKey]; !ok || val != v3.ExperimentalFeatureValue {
+		objCopy.Annotations[v3.ExperimentalFeatureKey] = v3.ExperimentalFeatureValue
+	}
```

Comment:
I might be reading this wrong, but do we need the `if` here? It seems like, no matter what, the value of `objCopy.Annotations[v3.ExperimentalFeatureKey]` will always be set to `v3.ExperimentalFeatureValue`.

Reply:
```suggestion
	objCopy.Annotations[v3.ExperimentalFeatureKey] = v3.ExperimentalFeatureValue
```